relay-server/src/services/store.rs

```diff
@@ -865,6 +865,35 @@ impl StoreService {
             }
         };
 
+        if let Some(measurements) = &mut span.measurements {
+            measurements.retain(|_, v| {
+                if let Some(v) = v {
+                    v.value.map_or(false, f64::is_finite)
+                } else {
+                    false
+                }
```

Comment:
Not sure if this is better:

```suggestion
                v.as_ref()
                    .and_then(|v| v.value)
                    .map_or(false, f64::is_finite)
```